src/pages/index.js

```diff
@@ -159,12 +187,77 @@ function closeModal(modal) {
 /* Anytime you pass a function to the evenListener, the even object is passed as the first arguement*/
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  /*earlier we set the value to the text content, now we're assigning text.content to the value*/
-  profileName.textContent = editModalNameInput.value;
-  /*now that we did it got profileName, let's do it for profileDescription*/
-  profileDescription.textContent = editModalDescriptionInput.value;
-  /*This code allows the editProfileContatiner to close once clicking submit*/
-  closeModal(editModal);
+  //Change text content to "Saving..."
+  //evt.submitter will target the submit button that corresponds to the form that was just submitted
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  //setButtonText(submitBtn, true);
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      //TODO: Use data argument instead of the input values
+      /*earlier we set the value to the text content, now we're assigning text.content to the value*/
+      profileName.textContent = data.name;
+      /*now that we did it got profileName, let's do it for profileDescription*/
+      profileDescription.textContent = data.about;
+      /*This code allows the editProfileContatiner to close once clicking submit*/
+      closeModal(editModal);
+    })
+    .catch((err) => {
+      console.error("Error updating user info", err);
+    })
+    //CHANGE TEXT CONTENT BACK TO "SAVE"
+    //finally will run after all API requests (whether failure or success)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+//6. Delete a card - Submission Button
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  //Per code reviewer: changing button text to "Deleting..." after API request to delete
+  submitBtn.textContent = "Deleting...";
+  api
+    .deleteCard(selectedCardId) //HAVE TO PASS THE ID(OTHERWISE WILL GET 404
+    .then(() => {
+      //TODO: remove the card & close the modal
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error);
```

Comment:
If you change the button text, then you need to **return** the initial button text back in `.finally`


```js
    })
    .catch(console.error)
    .finally(() => {
      submitBtn.textContent = "Delete";
    });
```